File: Electronics_Shop_17.Model/Requests/UpdateCustomerCoupon.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Electronics_Shop_17.Model.Requests
{
    public class UpdateCustomerCoupon
    {
        public int? CustomerId { get; set; }
        public int? CouponId { get; set; }
        public int? UsageCount { get; set; }
    }
}

File: angular17_web/src/app/models/customer-coupon/update.customer.coupon.ts
export class UpdateCustomerCoupon {
  customerId?: number;
  couponId?: number;
  usageCount?: number;

  constructor(data?: any) {
    if (!data) return;

    this.customerId = data.customerId;
    this.couponId = data.couponId;
    this.usageCount = data.usageCount;
  }
}
